### ISSUE TITLE
Duplicates in the URL typeahead

### ISSUE THREAD

**Issue by paulhitz** [paulhitz/yet-another-rest-client#14]
Since the URL typeahead gets its data from favorites, it can easily have duplicate URLs.

Need to filter these out somehow without breaking the pass by reference. See TODO in app.ctrl.js.

See also the requestUrl input typeahead attribute in main.html.

**Comment by Ashley-Waldron** [comment]
Is the solution for this to just stop the user being able to add favourites with duplicate request data?

**Comment by paulhitz** [comment]
No, it's valid for favorites to potentially have duplicate URLs. They might have different request headers.

However, there's no value to having them in the typeahead since it has no concept of request headers.

The solution is likely related to this line in main.html...

```
<input id="requestUrl" type="url" ng-model="requestUrl" typeahead="favorite.url for favorite in favorites | filter:{url:$viewValue} | limitTo:8" class="form-control" placeholder="http://www.example.com/resource/123" autofocus>
```

It (the filter part) should ignore duplicate URLs.
